boards/DIM/Src/App/App_DimWorld.c

```diff
@@ -76,6 +78,23 @@ void App_DimWorld_Destroy(struct DimWorld *world)
     free(world);
 }
 
+uint32_t App_DimWorld_GetCurrentSecond(struct DimWorld *world)
+{
+    return world->current_ms / 1000;
```

Comment:
Not sure how this will get truncated, might want to write it like:
``` suggestion
    return (uint32_t)((float)(world->current_ms) / 1000.0);
```
just to be safe

Reply:
That's fine with me, i think i will use `floor` for this and write some test to make sure what's specified in the jdocs is true (i.e. 999ms rounds down to 0, and 1001ms rounds down to 1s).

Another thought is that `getCurrentTimeInSeconds` won't be perfect when it overflows because the last second will be shorter. But I guess for our use-case we will never leave the MCU on for that long anyway so no need to cover for that use-case?

Reply:
I can just document in the jdocs that this method is imperfect during an overflow?

Reply:
Another update, turns out using floating point will cause some rounding errors which I noticed while writing the test `check_current_time_overflow`! I could go into more detail but basically the root cause is that floating point can express all integer values exactly.

I stuck with using unsigned integers and wrote tests to make sure the behaviour is correct

Reply:
I'm cool with it as long as we have carefully thought out test coverage.